Instrument all external links with Headlights API

DivyaPrabhakar opened the pull request:
References issue #1648 

Added calls to the Headlights API for the Introduction screen (Twitter, Github, and Forum) and the menu (Twitter, HelpX, and Forum) 

If we choose to merge this as is, then we need to update this wiki with the new headlights trio that is being used: https://github.com/adobe-photoshop/spaces-design/wiki/Headlights-Categories,-Subcategories,-and-Events

DivyaPrabhakar commented:
@iwehrman @mcilroyc There are no errors on this branch but I need help verifying whether the call to the API is getting logged or not. Let me know if you guys know of a way to make sure of this.

DivyaPrabhakar commented:
Also one question I have it: 

In src/js/util/ui.js:111 and src/js/actions/menu.js:131, do we need to make ```name``` and ```payload.category```, ```payload.subcategory```, ```payload.eventName``` required?

iwehrman commented on src/js/jsx/help/FirstLaunch.jsx:
```diff
@@ -33,7 +33,7 @@ define(function (require, exports, module) {
         ui = require("js/util/ui"),
         SVGIcon = require("jsx!js/jsx/shared/SVGIcon");
-        
+                
```
:space_invader:

iwehrman commented on src/js/actions/menu.js:
```diff
@@ -133,6 +133,8 @@ define(function (require, exports) {
             return Promise.reject(error);
         }
+        
+        headlights.logEvent(payload.category, payload.subcategory, payload.eventName);
```
These extra parameters should be documented in the JSDoc.

iwehrman commented:
@DivyaPrabhakar Looks good besides a few nits. If we decided that the parameters to those actions are "required" then would other changes need to be made? We should avoid logging empty strings. So let's make the smallest additional change to enforce that.

DivyaPrabhakar commented:
@iwehrman Currently, the only calls to menu.openURL are in menu-actions.json for the Forum, helpX, and Twitter. There are no empty strings being logged as of now and I added the JSDoc to show that url, category, subcategory, and eventName are required so as of now so hopefully this will be indication in the future. I also added a small check to only log the event if all three fields are not null.

iwehrman commented:
Looks great, thanks!
